apps/labs/posts/polars-plugins-lets-make-them-easier-to-use.md

```diff
@@ -0,0 +1,218 @@
+---
+title: "Polars Plugins: let's make them easier to use"
+description: 'Increasing ease-of-use of Polars plugins by improving an existing tutorial.'
+published: September 30, 2024
+authors: [bruno-kind]
+category: [OSS Experience, Internship]
+featuredImage:
+  src: /posts/polars-plugins-lets-make-them-easier-to-use/featured.png
+  alt: 'Polar bear with a yellow safety helmet and a red hammer on the ground, representing work-in-progress.'
+hero:
+  imageSrc: /posts/polars-plugins-lets-make-them-easier-to-use/hero.png
+  imageAlt: 'Polar bear with a yellow safety helmet and a red hammer on the ground, with the letters W.I.P., for work-in-progress.'
+---
+
+I am Bruno Kind, and this is the place where I tell the world about my journey in Open Source during July-September 2024.
+How I made it through an internship to work with Open Source, in an entry-level Rust position – yes, they're real!
+And how I managed to contribute and improve Marco Gorelli's existing [Polars Plugin Tutorial](https://marcogorelli.github.io/polars-plugins-tutorial/).
+Hop in, grab your popcorn, fasten your seatbelts.
+
+---
+
+- [What are Polars plugins?](#what-are-polars-plugins)
+- [Prelude and first week](#prelude-and-first-week)
+- [Writing plugins and chapters](#writing-plugins-and-chapters)
+- [Working at Quansight Labs](#working-at-quansight-labs)
+- [New ideas to make plugins more approachable](#new-ideas-to-make-plugins-more-approachable)
+- [The end](#the-end)
+
+---
+
+## What are Polars plugins?
+
+> Expression plugins are the preferred way to create user defined functions. They allow you to compile a Rust function and register that as an expression into the Polars library. - [Official docs](https://docs.pola.rs/user-guide/expressions/plugins/)
+
+User defined functions are quite important in the context of DataFrames. Polars is made in Rust and distributed as a library for Python, R and NodeJS (besides Rust itself). Users of these other languages might be tempted to write User Defined Functions (UDFs) in these languages, since it's how they're using Polars, not to mention the comfort zone factor. However, plugins are a way to possibly increase the performance of these functions by orders of magnitude! By writing UDFs in Rust (as expression plugins), Polars is able to work more efficiently than if they were defined in the languages mentioned above. Expression plugins are _a_ type of plugin, and the main target of this article - whenver you read plugin, I'm referring to expression plugins, unless stated otherwise. The other type is called an IO plugin, more on that later.
```

Comment:
```suggestion
User defined functions are quite important in the context of DataFrames. Polars is made in Rust and distributed as a library for Python, R and NodeJS (besides Rust itself). Users of these other languages might be tempted to write User Defined Functions (UDFs) in these languages, since it's how they're using Polars, not to mention the comfort zone factor. However, plugins are a way to possibly increase the performance of these functions by orders of magnitude! By writing UDFs in Rust (as expression plugins), Polars is able to work more efficiently than if they were defined in the languages mentioned above. Expression plugins are _a_ type of plugin, and the main target of this article - whenever you read plugin, I'm referring to expression plugins, unless stated otherwise. The other type is called an IO plugin, more on that later.
```